In [11]:
import os
import json
import pandas as pd

baseline_path = "/home/xuyang1/mle-bench-aide-gpt-4o-2024-mini"
output_file = "/home/xuyang1/mle-bench-aide-gpt-4o-2024-mini/res_gpt_4o_aide.csv"

data_list = []

competition_score_dict = {}

for folder in os.listdir(baseline_path):
    folder_path = os.path.join(baseline_path, folder)
    if os.path.isdir(folder_path):
        grading_file_path = None
        for file in os.listdir(folder_path):
            if file.endswith("grading_report.json"):
                grading_file_path = os.path.join(folder_path, file)
                break

        with open(grading_file_path, "r") as f:
            grading_data = json.load(f)

            for report in grading_data.get("competition_reports", []):
                data_list.append({
                    "competition_id": report.get("competition_id"),
                    "score": report.get("score"),
                    "gold_threshold": report.get("gold_threshold"),
                    "silver_threshold": report.get("silver_threshold"),
                    "bronze_threshold": report.get("bronze_threshold"),
                    "median_threshold": report.get("median_threshold"),
                    "any_medal": report.get("any_medal"),
                    "gold_medal": report.get("gold_medal"),
                    "silver_medal": report.get("silver_medal"),
                    "bronze_medal": report.get("bronze_medal"),
                    "above_median": report.get("above_median"),
                    "submission_exists": report.get("submission_exists"),
                    "valid_submission": report.get("valid_submission"),
                })
                competition_score_dict[report.get("competition_id")] = (report.get("score"), report.get("is_lower_better"))
competition_df = pd.DataFrame(data_list)
# competition_df


In [12]:
# import bo-full result
bo_full_path = "runs/2025-01-22T08-01-25-GMT_run-group_aide-bo-full/2025-01-27T12-47-40-GMT_grading_report.json"
with open(bo_full_path, "r") as f:
    bo_full_data = json.load(f)

data_list_bo = []

competition_score_dict_bo = {}
for report in bo_full_data.get("competition_reports", []):
    data_list_bo.append({
        "competition_id": report.get("competition_id"),
        "score": report.get("score"),
        "gold_threshold": report.get("gold_threshold"),
        "silver_threshold": report.get("silver_threshold"),
        "bronze_threshold": report.get("bronze_threshold"),
        "median_threshold": report.get("median_threshold"),
        "any_medal": report.get("any_medal"),
        "gold_medal": report.get("gold_medal"),
        "silver_medal": report.get("silver_medal"),
        "bronze_medal": report.get("bronze_medal"),
        "above_median": report.get("above_median"),
        "submission_exists": report.get("submission_exists"),
        "valid_submission": report.get("valid_submission"),
    })
    competition_score_dict_bo[report.get("competition_id")] = (report.get("score"), report.get("is_lower_better"))
bo_full_df = pd.DataFrame(data_list_bo)
# bo_full_df


In [3]:
# only use the lite competition id
competition_id_lite = bo_full_df['competition_id'].unique()
competition_df = competition_df[competition_df['competition_id'].isin(competition_id_lite)]


In [4]:
grouped_df = competition_df.groupby('competition_id').agg(
    total_competitions=('competition_id', 'size'),
    valid_submissions=('valid_submission', 'sum'),
    submission_exists=('submission_exists', 'sum'),
    any_medal=('any_medal', 'sum'),
    gold_medal=('gold_medal', 'sum'),
    silver_medal=('silver_medal', 'sum'),
    bronze_medal=('bronze_medal', 'sum'),
    above_median=('above_median', 'sum')
).reset_index()
# grouped_df.to_csv("/home/xuyang1/mle-bench-aide-gpt-4o-2024-mini/res_gpt_4o_aide_each.csv", index=False)
# grouped_df

grouped_df_bo = bo_full_df.groupby('competition_id').agg(
    total_competitions=('competition_id', 'size'),
    valid_submissions=('valid_submission', 'sum'),
    submission_exists=('submission_exists', 'sum'),
    any_medal=('any_medal', 'sum'),
    gold_medal=('gold_medal', 'sum'),
    silver_medal=('silver_medal', 'sum'),
    bronze_medal=('bronze_medal', 'sum'),
    above_median=('above_median', 'sum')
).reset_index()
# grouped_df_bo

In [5]:
bo_full_df

competition_id    score  gold_threshold  \
0                      aerial-cactus-identification  0.99792        1.000000   
1                     aptos2019-blindness-detection      NaN        0.930508   
2            detecting-insults-in-social-commentary      NaN        0.833210   
3                          dog-breed-identification      NaN        0.000500   
4                         denoising-dirty-documents  0.21055        0.017940   
5                               leaf-classification  0.86964        0.000000   
6                  histopathologic-cancer-detection      NaN        0.983500   
7                dogs-vs-cats-redux-kernels-edition      NaN        0.038820   
8     jigsaw-toxic-comment-classification-challenge  0.96861        0.987400   
9                                   mlsp-2013-birds  0.83193        0.935270   
10                             random-acts-of-pizza  0.59801        0.979080   
11               new-york-city-taxi-fare-prediction  4.72937        2.833770   
12                       plant-pathology-2020-fgvc7  0.95100        0.978360   
13         nomad2018-predict-transparent-conductors  0.06297        0.055890   
14         ranzcr-clip-catheter-line-classification      NaN        0.973570   
15               tabular-playground-series-may-2022  0.89260        0.998230   
16                     spooky-author-identification  0.41671        0.165060   
17               tabular-playground-series-dec-2021  0.95836        0.956600   
18                siim-isic-melanoma-classification      NaN        0.945500   
19    text-normalization-challenge-english-language  0.93299        0.997240   
20    text-normalization-challenge-russian-language  0.97348        0.990120   
21  the-icml-2013-whale-challenge-right-whale-redux  0.82432        0.989610   

    silver_threshold  bronze_threshold  median_threshold  any_medal  \
0           1.000000          1.000000          0.999100      False   
1           0.919654          0.914492          0.888912      False   
2           0.823070          0.791110          0.778420      False   
3           0.005390          0.045980          0.472050      False   
4           0.026090          0.045170          0.073250      False   
5           0.007910          0.015260          0.108345      False   
6           0.979800          0.973800          0.947700      False   
7           0.050380          0.061270          0.122160      False   
8           0.986680          0.986390          0.980790      False   
9           0.900380          0.873720          0.865720      False   
10          0.764820          0.692100          0.599595      False   
11          2.881910          2.923710          3.597410      False   
12          0.974650          0.973610          0.948520      False   
13          0.062290          0.065820          0.069880       True   
14          0.971520          0.970900          0.967500      False   
15          0.998220          0.998180          0.972675      False   
16          0.269960          0.293810          0.418785      False   
17          0.956580          0.956580          0.953420       True   
18          0.940100          0.937000          0.912800      False   
19          0.991350          0.990380          0.990370      False   
20          0.982320          0.975920          0.975910      False   
21          0.950170          0.905210          0.865210      False   

    gold_medal  silver_medal  bronze_medal  above_median  submission_exists  \
0        False         False         False         False               True   
1        False         False         False         False              False   
2        False         False         False         False               True   
3        False         False         False         False              False   
4        False         False         False         False               True   
5        False         False         False         False               True   
6        

In [6]:


# Convert counts to percentages
grouped_df['valid_submissions'] = (grouped_df['valid_submissions'] / grouped_df['total_competitions']) * 100
grouped_df['submission_exists'] = (grouped_df['submission_exists'] / grouped_df['total_competitions']) * 100
grouped_df['any_medal'] = (grouped_df['any_medal'] / grouped_df['total_competitions']) * 100
grouped_df['gold_medal'] = (grouped_df['gold_medal'] / grouped_df['total_competitions']) * 100
grouped_df['silver_medal'] = (grouped_df['silver_medal'] / grouped_df['total_competitions']) * 100
grouped_df['bronze_medal'] = (grouped_df['bronze_medal'] / grouped_df['total_competitions']) * 100
grouped_df['above_median'] = (grouped_df['above_median'] / grouped_df['total_competitions']) * 100
# grouped_df.to_csv("/home/xuyang1/mle-bench-aide-gpt-4o-2024-mini/res_gpt_4o_aide_mini_each2.csv", index=False)
# grouped_df


grouped_df_bo['valid_submissions'] = (grouped_df_bo['valid_submissions'] / grouped_df_bo['total_competitions']) * 100
grouped_df_bo['submission_exists'] = (grouped_df_bo['submission_exists'] / grouped_df_bo['total_competitions']) * 100
grouped_df_bo['any_medal'] = (grouped_df_bo['any_medal'] / grouped_df_bo['total_competitions']) * 100
grouped_df_bo['gold_medal'] = (grouped_df_bo['gold_medal'] / grouped_df_bo['total_competitions']) * 100
grouped_df_bo['silver_medal'] = (grouped_df_bo['silver_medal'] / grouped_df_bo['total_competitions']) * 100
grouped_df_bo['bronze_medal'] = (grouped_df_bo['bronze_medal'] / grouped_df_bo['total_competitions']) * 100
grouped_df_bo['above_median'] = (grouped_df_bo['above_median'] / grouped_df_bo['total_competitions']) * 100
# grouped_df_bo.to_csv("/home/xuyang1/mle-bench-aide-gpt-4o-2024-mini/res_bo_full_each2.csv", index=False)
# grouped_df_bo


In [7]:
# Made Submission Rate

comp_made_rate = grouped_df['submission_exists'].sum() / len(grouped_df)
comp_made_rate_bo = grouped_df_bo['submission_exists'].sum() / len(grouped_df_bo)
print("comp_made_rate:", comp_made_rate)
print("comp_made_rate_bo:", comp_made_rate_bo,'\n')

# Valid Submission Rate
comp_finished_rate = grouped_df['valid_submissions'].sum() / len(grouped_df)
comp_finished_rate_bo = grouped_df_bo['valid_submissions'].sum() / len(grouped_df_bo)
print("comp_finished_rate:", comp_finished_rate)
print("comp_finished_rate_bo:", comp_finished_rate_bo,'\n')

# Above Median Rate
above_median_rate = grouped_df['above_median'].sum() / len(grouped_df)
above_median_rate_bo = grouped_df_bo['above_median'].sum() / len(grouped_df_bo)
print("above_median_rate:", above_median_rate)
print("above_median_rate_bo:", above_median_rate_bo,'\n')

# Gold Medal Rate
gold_medal_rate = grouped_df['gold_medal'].sum() / len(grouped_df)
gold_medal_rate_bo = grouped_df_bo['gold_medal'].sum() / len(grouped_df_bo)
print("gold_medal_rate:", gold_medal_rate)
print("gold_medal_rate_bo:", gold_medal_rate_bo,'\n')

# Silver Medal Rate
silver_medal_rate = grouped_df['silver_medal'].sum() / len(grouped_df)
silver_medal_rate_bo = grouped_df_bo['silver_medal'].sum() / len(grouped_df_bo)
print("silver_medal_rate:", silver_medal_rate)
print("silver_medal_rate_bo:", silver_medal_rate_bo,'\n')

# Bronze Medal Rate
bronze_medal_rate = grouped_df['bronze_medal'].sum() / len(grouped_df)
bronze_medal_rate_bo = grouped_df_bo['bronze_medal'].sum() / len(grouped_df_bo)
print("bronze_medal_rate:", bronze_medal_rate)
print("bronze_medal_rate_bo:", bronze_medal_rate_bo,'\n')

# Any Medal Rate
any_medal_rate = grouped_df['any_medal'].sum() / len(grouped_df)
any_medal_rate_bo = grouped_df_bo['any_medal'].sum() / len(grouped_df_bo)
print("any_medal_rate:", any_medal_rate)
print("any_medal_rate_bo:", any_medal_rate_bo,'\n')


comp_made_rate: 59.09090909090909
comp_made_rate_bo: 72.72727272727273 

comp_finished_rate: 50.0
comp_finished_rate_bo: 68.18181818181819 

above_median_rate: 18.181818181818183
above_median_rate_bo: 18.181818181818183 

gold_medal_rate: 4.545454545454546
gold_medal_rate_bo: 4.545454545454546 

silver_medal_rate: 4.545454545454546
silver_medal_rate_bo: 0.0 

bronze_medal_rate: 0.0
bronze_medal_rate_bo: 4.545454545454546 

any_medal_rate: 9.090909090909092
any_medal_rate_bo: 9.090909090909092 



In [8]:
markdown_table = f"""
| Model      | Made Submission (%) | Valid Submission (%) | Above Median (%) | Bronze (%) | Silver (%) | Gold (%) | Any Medal (%) |
|------------|---------------------|-----------------------|------------------|------------|------------|----------|---------------|
| AIDE (Ours, gpt-4o_2024-08-06) | {comp_made_rate:.1f}             | {comp_finished_rate:.1f}               | {above_median_rate:.1f}           | {bronze_medal_rate:.1f}      | {silver_medal_rate:.1f}      | {gold_medal_rate:.1f}    | {any_medal_rate:.1f}         |
| AIDE-BO (Ours, gpt-4o_2024-08-06) | {comp_made_rate_bo:.1f}             | {comp_finished_rate_bo:.1f}               | {above_median_rate_bo:.1f}           | {bronze_medal_rate_bo:.1f}      | {silver_medal_rate_bo:.1f}      | {gold_medal_rate_bo:.1f}    | {any_medal_rate_bo:.1f}         |
"""

print(markdown_table)


| Model      | Made Submission (%) | Valid Submission (%) | Above Median (%) | Bronze (%) | Silver (%) | Gold (%) | Any Medal (%) |
|------------|---------------------|-----------------------|------------------|------------|------------|----------|---------------|
| AIDE (Ours, gpt-4o_2024-08-06) | 59.1             | 50.0               | 18.2           | 0.0      | 4.5      | 4.5    | 9.1         |
| AIDE-BO (Ours, gpt-4o_2024-08-06) | 72.7             | 68.2               | 18.2           | 4.5      | 0.0      | 4.5    | 9.1         |



| Model      | Made Submission (%) | Valid Submission (%) | Above Median (%) | Bronze (%) | Silver (%) | Gold (%) | Any Medal (%) |
|------------|---------------------|-----------------------|------------------|------------|------------|----------|---------------|
| AIDE (OpenAI, gpt-4o_2024-08-06) | 82.8             | 66.7               | 30.4           | 3.4      | 6.0      | 9.6    | 19.0         |
| AIDE (Ours, gpt-4o_2024-08-06) | 59.1             | 50.0               | 18.2           | 0.0      | 4.5      | 4.5    | 9.1         |
| AIDE-BO (Ours, gpt-4o_2024-08-06) | 72.7             | 66.7               | 18.2           | 0.0      | 4.5      | 4.5    | 9.1         |

In [16]:
# Create a table comparing scores for competitions with valid submissions
valid_comps = grouped_df[grouped_df['valid_submissions'] > 0]['competition_id'].unique()

if_BO_help_improve = []
for comp in valid_comps:
    if competition_score_dict[comp][0] > competition_score_dict_bo[comp][0]:
        if competition_score_dict_bo[comp][1]:
            if_BO_help_improve.append(True)
        else:
            if_BO_help_improve.append(False)
    else:
        if competition_score_dict_bo[comp][1]:
            if_BO_help_improve.append(False)
        else:
            if_BO_help_improve.append(True)

score_comparison = pd.DataFrame({
    'Competition': valid_comps,
    # 'AIDE (OpenAI)': [0.0] * len(valid_comps),  # Placeholder since we don't have OpenAI scores
    'AIDE (Ours)': [competition_score_dict[comp][0] for comp in valid_comps],
    'AIDE-BO': [competition_score_dict_bo[comp][0] for comp in valid_comps],
    'is_lower_better': [competition_score_dict_bo[comp][1] for comp in valid_comps],
    'BO help improve?': if_BO_help_improve
})

# Format scores to 4 decimal places
score_comparison['AIDE (Ours)'] = score_comparison['AIDE (Ours)'].map('{:.4f}'.format)
score_comparison['AIDE-BO'] = score_comparison['AIDE-BO'].map('{:.4f}'.format)

print("\nScore Comparison for Valid Submissions:")
print(score_comparison.to_string(index=False))



Score Comparison for Valid Submissions:
                                    Competition AIDE (Ours) AIDE-BO  is_lower_better  BO help improve?
                   aerial-cactus-identification      0.9998  0.9979            False             False
  jigsaw-toxic-comment-classification-challenge      0.9737  0.9686            False             False
                            leaf-classification      0.6788  0.8696             True             False
             new-york-city-taxi-fare-prediction      4.1015  4.7294             True             False
       nomad2018-predict-transparent-conductors      0.0603  0.0630             True             False
                           random-acts-of-pizza      0.5767  0.5980            False              True
                   spooky-author-identification      0.4130  0.4167             True             False
             tabular-playground-series-dec-2021      0.9584  0.9584            False              True
             tabular-playground-

In [15]:
if_BO_help_improve

[False, False, False, False]

In [16]:
grouped_df

competition_id  total_competitions  \
0                      aerial-cactus-identification                   1   
1                     aptos2019-blindness-detection                   1   
2                         denoising-dirty-documents                   1   
3            detecting-insults-in-social-commentary                   1   
4                          dog-breed-identification                   1   
5                dogs-vs-cats-redux-kernels-edition                   1   
6                  histopathologic-cancer-detection                   1   
7     jigsaw-toxic-comment-classification-challenge                   1   
8                               leaf-classification                   1   
9                                   mlsp-2013-birds                   1   
10               new-york-city-taxi-fare-prediction                   1   
11         nomad2018-predict-transparent-conductors                   1   
12                       plant-pathology-2020-fgvc7                   1   
13                             random-acts-of-pizza                   1   
14         ranzcr-clip-catheter-line-classification                   1   
15                siim-isic-melanoma-classification                   1   
16                     spooky-author-identification                   1   
17               tabular-playground-series-dec-2021                   1   
18               tabular-playground-series-may-2022                   1   
19    text-normalization-challenge-english-language                   1   
20    text-normalization-challenge-russian-language                   1   
21  the-icml-2013-whale-challenge-right-whale-redux                   1   

    valid_submissions  submission_exists  any_medal  gold_medal  silver_medal  \
0               100.0              100.0        0.0         0.0           0.0   
1                 0.0                0.0        0.0         0.0           0.0   
2                 0.0              100.0        0.0         0.0           0.0   
3                 0.0              100.0        0.0         0.0           0.0   
4                 0.0                0.0        0.0         0.0           0.0   
5                 0.0                0.0        0.0         0.0           0.0   
6                 0.0                0.0        0.0         0.0           0.0   
7               100.0              100.0        0.0         0.0           0.0   
8               100.0              100.0        0.0         0.0           0.0   
9                 0.0                0.0        0.0         0.0           0.0   
10              100.0              100.0        0.0         0.0           0.0   
11              100.0              100.0      100.0         0.0         100.0   
12                0.0                0.0        0.0         0.0           0.0   
13              100.0              100.0        0.0         0.0           0.0   
14                0.0                0.0        0.0         0.0           0.0   
15                0.0                0.0        0.0         0.0           0.0   
16              100.0              100.0        0.0         0.0           0.0   
17              100.0              100.0      100.0       100.0           0.0   
18              100.0              100.0        0.0         0.0           0.0   
19                0.0                0.0        0.0         0.0           0.0   
20              100.0              100.0        0.0         0.0           0.0   
21              100.0              100.0        0.0         0.0           0.0   

    bronze_medal  above_median  
0            0.0         100.0  
1            0.0           0.0  
2            0.0           0.0  
3            0.0           0.0  
4            0.0           0.0  
5            0.0           0.0  
6            0.0           0.0  
7            0.0           0.0  
8            0.0           0.0  
9            0.0           0.0  
10           0.0           0.0  
11           0.0         100.0  
12           0.0  